### DISCUSSION TITLE
SATp1 no transmission for specific bands

### DISCUSSION THREAD

**Discussion by jlashner** [simonsobs/sodetlib#413] [Issues and Debugging]
Sometimes on SATp1 we've seen that certain individual bands have no transmission. Because this is an individual band and not full AMC, I think we can rule out connection / cryo error. 

This results in transmission plots that look like this: 
![image](https://github.com/simonsobs/sodetlib/assets/4718487/07a9f830-dbdc-4cec-95a4-565259535a3f)
that end up causing all sorts of errors for various sodetlib operation issues.

Sometimes this can be fixed by hammering the slot. Most recently, I've found this behavior on Mv22: (more is documented in [this notes page](https://simonsobs.atlassian.net/wiki/spaces/SOPS/pages/282001416/2024-01-31+Smurf+recovery+Mv7+Mv22+Jack))

At first bands 3 and 5 were showing no transmission. After hammer, bands 3 and 5 were both fixed, but band 0 was showing no transmission. @swh76 do you have any idea what might be causing this?

**Comment by jlashner** [comment]
Just some more detail, after tuning finished I tried to take full-band response to see if that looked the same. Full band response for band 0 actually looks good: 
![image](https://github.com/simonsobs/sodetlib/assets/4718487/02318eb6-8dc8-48a8-92ea-4af5d4c18bdb)
while the full-band amplitude sweep looks awful: 
![image](https://github.com/simonsobs/sodetlib/assets/4718487/630271b6-291e-4f97-a1ec-c274cd1ec934)

This could be a clue.

**Comment by swh76** [comment]
@jlashner this one's new to me.
  First thoughts;

- The `full_band_resp` vs `full_band_ampl_sweep` is interesting.  The former just plays a waveform on the DAC and receives on the ADC for that band ; so it bypasses the DSP.  `full_band_ampl_sweep`, like `find_freqs`, generates and receives tones via the DSP (e.g. via the analysis/synthesis PFBs).  So may be pointing to a misconfiguration of the DSP.
- Can you get now or next time this happens, the following?
-- State dump.  I usually do this; `import time; S.save_state(f'/data/smurf_data/{int(time.time())}_slot{S.slot_number}_state.yml')`.  FYI it takes 30 sec or so before the file finishes getting written to disk.
-- Rogue (you call this streamer) logs ; here looking for any missed reg sets.  E.g. if for some reason you're dropping an occasional register write, could imagine you're missing a critical DSP configuration register.
-- Does issue resolve w/ second `setup`?

And a question ; does a band ever "go bad" e.g. `full_band_ampl_sweep` looks fine but stops working later?  Or as far as we know if it's good after initial hammer, does it stay good?
